### PR TITLE
Improve StorageInterface, add method `get` 

### PR DESCRIPTION
**Pull request by Enet4**
This PR proposes a backwards-compatible extension to `StorageInterface` with default method `get`. Unlike `at`, `get` is designed to retrieve up to a single item in storage, so it does not need file system tree traversal or building up iterables and iterators.

A default implementation is provided that uses `at` underneath, but implementors may wish to override it for more efficiency.

This PR also includes a few quality-of-life improvements, including some clarifications to other methods in the plugin interface.

### Summary

- Add default method `StorageInterface#get`, to retrieve a single item from storage
   - update code in core to use it where suitable
   - also correct image servlet to only query for instances to DIM providers if specified
- Fix example code in `at`
- Clarify in `list` that a trailing forward slash means that it's a directory

### CODE REVIEW

**Review by bastiao — APPROVED** [review]
I understand that it useful. 
I will merge it soon.

**Comment by Enet4** [comment]
@bastiao Since the next Dicoogle version is intended to be a patch version, let's keep this PR on hold until we plan to deploy a feature release.

File: sdk/src/main/java/pt/ua/dicoogle/sdk/StorageInterface.java [hunk 1]
```diff
@@ -62,18 +62,56 @@ public default boolean handles(URI location) {
      * 
      * The provided scheme is not relevant at this point, but the developer must avoid calling this method
      * with a path of a different schema.
```
Comment (Enet4) [inline sdk/src/main/java/pt/ua/dicoogle/sdk/StorageInterface.java:64]:
```suggestion
     * with a path of a different scheme.
```
```suggestion
     * with a path of a different schema.
```

File: sdk/src/main/java/pt/ua/dicoogle/sdk/StorageInterface.java [hunk 2]
```diff
@@ -62,18 +62,56 @@ public default boolean handles(URI location) {
...
      * @see StorageInputStream
      */
     public Iterable<StorageInputStream> at(URI location, Object... parameters);
 
+    /**
+     * Obtains an item stored at the exact location specified.
+     *
+     * The provided scheme is not relevant at this point,
+     * but the developer must avoid calling this method
+     * with a path of a different schema.
```
Comment (Enet4) [inline sdk/src/main/java/pt/ua/dicoogle/sdk/StorageInterface.java:85]:
```suggestion
     * with a path of a different scheme.
```
```suggestion
     * with a path of a different schema.
```

**Review by Enet4 — COMMENTED** [review]
I might as well fix this typo.

**Review by tiberio-baptista — APPROVED** [review]
The changes are quite nice, and current behavior of these components remains functional. LGTM!

**Comment by Enet4** [comment]
@bastiao Next version of Dicoogle should be 3.4.0, so we're ready to start merging this and the other ones in the milestone:

- #660
- #598
- #695
- Maybe even #631

**Pull request merged**